composer/trainer/trainer.py

```diff
@@ -1698,9 +1698,9 @@ def _train_batch(self, use_grad_scaling: bool):
                 else:
                     original_grad_accum = self.state.grad_accum
                     self.state.grad_accum = min(2 * self.state.grad_accum, device_batch_size)
-                    log.info(('CUDA out of memory detected. Gradient Accumulation '
-                              f'increased from {original_grad_accum} -> {self.state.grad_accum}, '
-                              'and the batch will be retrained.'))
+                    warnings.warn(('CUDA out of memory detected. Gradient Accumulation '
```

Comment:
nit: make a RuntimeWarning (or another subclass of Warning) as it's not something the user should have to deal with.

```suggestion
                    warnings.warn(RuntimeWarning('CUDA out of memory detected. Gradient Accumulation '
```